packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart

```diff
@@ -0,0 +1,320 @@
+  }) {
+    return AndroidWebViewControllerCreationParams(
+      androidWebViewProxy: androidWebViewProxy,
+      androidWebStorage:
+          androidWebStorage ?? android_webview.WebStorage.instance,
+    );
+  }
+
+  /// Handles constructing objects and calling static methods for the Android WebView
+  /// native library.
+  @visibleForTesting
+  final AndroidWebViewProxy androidWebViewProxy;
+
+  /// Manages the JavaScript storage APIs provided by the [android_webview.WebView].
+  @visibleForTesting
+  final android_webview.WebStorage androidWebStorage;
+}
+
+/// Implementation of the [PlatformWebViewController] with the Android WebView API.
+class AndroidWebViewController extends PlatformWebViewController {
+  /// Creates a new [AndroidWebViewCookieManager].
+  AndroidWebViewController(PlatformWebViewControllerCreationParams params)
+      : super.implementation(params is AndroidWebViewControllerCreationParams
+            ? params
+            : AndroidWebViewControllerCreationParams
+                .fromPlatformWebViewControllerCreationParams(params));
+
+  AndroidWebViewControllerCreationParams get _androidWebViewParams =>
+      params as AndroidWebViewControllerCreationParams;
+
+  /// The native [android_webview.WebView] being controlled.
+  late final android_webview.WebView _webView = withWeakRefenceTo(this, (
+    WeakReference<AndroidWebViewController> weakReference,
+  ) {
+    return _androidWebViewParams.androidWebViewProxy.createAndroidWebView(
+      useHybridComposition: true,
+    );
+  });
+
+  /// The native [android_webview.FlutterAssetManager] allows managing assets.
+  late final android_webview.FlutterAssetManager _flutterAssetManager =
+      withWeakRefenceTo(this, (
+    WeakReference<AndroidWebViewController> weakReference,
+  ) {
+    return _androidWebViewParams.androidWebViewProxy
+        .createFlutterAssetManager();
+  });
+
+  final Map<String, AndroidJavaScriptChannelParams> _javaScriptChannelParams =
+      <String, AndroidJavaScriptChannelParams>{};
+
+  @override
+  Future<void> loadFile(
+    String absoluteFilePath,
+  ) {
+    final String url = absoluteFilePath.startsWith('file://')
+        ? absoluteFilePath
+        : 'file://$absoluteFilePath';
+
+    _webView.settings.setAllowFileAccess(true);
+    return _webView.loadUrl(url, <String, String>{});
+  }
+
+  @override
+  Future<void> loadFlutterAsset(
+    String key,
+  ) async {
+    final String assetFilePath =
+        await _flutterAssetManager.getAssetFilePathByName(key);
+    final List<String> pathElements = assetFilePath.split('/');
+    final String fileName = pathElements.removeLast();
+    final List<String?> paths =
+        await _flutterAssetManager.list(pathElements.join('/'));
+
+    if (!paths.contains(fileName)) {
+      throw ArgumentError(
+        'Asset for key "$key" not found.',
+        'key',
+      );
+    }
+
+    return _webView.loadUrl(
+      'file:///android_asset/$assetFilePath',
```

Comment:
We should do the same here; `Uri.file('/android_asset/$assetFilePath')`.

Reply:
Fixed.